en, zh: add docs of topology spread constraints and security context (#1184)
pingcap/docs-tidb-operator: en/deploy-tidb-dm.md

```

### Topology Spread Constraint

> Use `topologySpreadConstraints` can make pods even spread in different topology. See [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/)

Reviewer: ```suggestion
By configuring `topologySpreadConstraints`, you can make pods evenly spread in different topologies. For instructions about configuring `topologySpreadConstraints`, see [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/).
```

> By configuring `topologySpreadConstraints`, you can make pods evenly spread in different topologies. For instructions about configure `topologySpreadConstraints`, see [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/).

Reviewer: ```suggestion
By configuring `topologySpreadConstraints`, you can make pods evenly spread in different topologies. For instructions about configuring `topologySpreadConstraints`, see [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/).
```

> **Note:**
>

> > EvenPodsSpread feature gate must be enabled. It has no effect on Kubernetes before v1.16 or which disables this feature gate.

Reviewer: ```suggestion
> To use `topologySpreadConstraints`, you must enable the `EvenPodsSpread` feature gate. If the Kubernetes version in use is earlier than v1.16 or if the `EvenPodsSpread` feature gate is disabled, the configuration of `topologySpreadConstraints` does not take effect.
```

> `topologySpreadConstraints` can be set at cluster level (`spec.topologySpreadConstraints`) for all of components or component level (e.g. `spec.tidb.topologySpreadConstraints`) for specific component.

Reviewer: ```suggestion
You can either configure `topologySpreadConstraints` at a cluster level (`spec.topologySpreadConstraints`) for all components or at a component level (such as `spec.tidb.topologySpreadConstraints`) for specific components.
```

> This is an example：

Reviewer: ```suggestion
The following is an example configuration：
```

> You can either configure `topologySpreadConstraints` at a cluster level (`spec.topologySpreadConstraints`) for all of components or at a component level (such as `spec.tidb.topologySpreadConstraints`) for a specific component.

Reviewer: ```suggestion
You can either configure `topologySpreadConstraints` at a cluster level (`spec.topologySpreadConstraints`) for all components or at a component level (such as `spec.tidb.topologySpreadConstraints`) for specific components.
```

{{< copyable "" >}}

```yaml
topologySpreadConstrains:
- topologyKey: kubernetes.io/hostname
- topologyKey: topology.kubernetes.io/zone
```

> Config above can make pods of same component even spread on different zones and nodes.

Reviewer: ```suggestion
The example configuration can make pods of the same component evenly spread on different zones and nodes.
```

> Now `topologySpreadConstraints` only support `topologyKey` field. Config above will expand as below in pod spec.

Reviewer: ```suggestion
Currently, `topologySpreadConstraints` only supports the configuration of the `topologyKey` field. In the pod spec file, the above example configuration will be automatically expanded as follows:
```

```yaml
topologySpreadConstrains:
- topologyKey: kubernetes.io/hostname
  maxSkew: 1
  whenUnsatisfiable: DoNotSchedule
  labelSelector: <object>
- topologyKey: topology.kubernetes.io/zone
  maxSkew: 1
  whenUnsatisfiable: DoNotSchedule
  labelSelector: <object>
```

> **Note:**
>

> > This feature can be used to replace [TiDB Scheduler](tidb-scheduler.md) for even scheduling.

Reviewer: ```suggestion
> You can use this feature to replace [TiDB Scheduler](tidb-scheduler.md) for evenly scheduling.
```
